Test Tone from Command Line Not Working

Running 2.0 Beta 4

I don't know if I'm doing something wrong, but I can't get either "-t" or "--test-tone" to work anymore.

```
pi@WSPRPi:~ $ sudo wsprrypi -t 10140200
[INFO ] WsprryPi version 2.0_Beta.4-2-0-Beta (2-0-Beta).
[INFO ] Running on: Raspberry Pi 3 Model B Plus Rev 1.4.
[INFO ] Process PID: 881
[INFO ] WSPR loop running.
[ERROR] Empty or invalid frequency list; disabling transmission.
[INFO ] A test tone will be generated at 10.140200 MHz .
```

```
pi@WSPRPi:~ $ sudo wsprrypi --test-tone 10140200
[INFO ] WsprryPi version 2.0_Beta.4-2-0-Beta (2-0-Beta).
[INFO ] Running on: Raspberry Pi 3 Model B Plus Rev 1.4.
[INFO ] Process PID: 877
[INFO ] WSPR loop running.
[ERROR] Empty or invalid frequency list; disabling transmission.
[INFO ] A test tone will be generated at 10.140200 MHz .
```

Note the entry: "[ERROR] Empty or invalid frequency list; disabling transmission."

No signal is generated and I have to hit ctrl-c to get the bash command prompt back.

--- a/release_tools/developer_notes.md
+++ b/release_tools/developer_notes.md
@@ -169,22 +169,25 @@ sudo systemctl stop wsprrypi
 sudo systemctl disable wsprrypi
 ```
 
-## Required Libs
+## Required Devel Libs
 
-If you did not run `install.sh` from within the Wsprry Pi repo or with the WsprryPi curl command, you would need some libs to compile the project:
+If you did not run `install.sh` from within the Wsprry Pi repo or with the WsprryPi curl command, will need some libs to execute the project:
 
-- jq
-- git
+
 - apache2
 - php
-- gpiod
-- libgpiod-dev
 - chrony
+- libgpiod2
+
+If you are doing development, you will need, in additon:
+
+- git
+- libgpiod-dev
 
 Install these with:
 
 ``` bash
-sudo apt install jq git apache2 php chrony gpiod libgpiod-dev -y
+sudo apt install git libgpiod-dev apache2 php chrony libgpiod2 -y
 ```
 
 ## A Note About Submodules

--- a/src/arg_parser.cpp
+++ b/src/arg_parser.cpp
@@ -149,7 +149,6 @@ void callback_ini_changed()
     {
         if (config.transmit)
         {
-            // TODO:  Make sure config.transmit is changed before we make this determination
             // Transmit not changed, make pending change
             llog.logS(INFO, "INI file changed, reload after transmission.");
         }
@@ -475,15 +474,9 @@ bool validate_config_data()
     // Handle test tone mode (TONE mode does not require callsign, grid, etc.)
     if (config.mode == ModeType::TONE)
     {
-        if (config.test_tone <= 0.0)
-        {
-            llog.logE(FATAL, "Test tone frequency must be positive.");
-            return false;
-        }
-
         // Log test tone frequency
-        llog.logS(INFO, "A test tone will be generated at",
-                  lookup.freq_display_string(config.test_tone), ".");
+        llog.logS(INFO, "A test tone will be generated at:",
+                  lookup.freq_display_string(config.test_tone));
     }
     else if (config.mode == ModeType::WSPR)
     {
@@ -652,7 +645,7 @@ bool set_frequencies()
     }
 
     // Ensure we have at least one valid frequency.
-    if (config.center_freq_set.empty()) {
+    if (config.center_freq_set.empty() && config.mode == ModeType::WSPR) {
         llog.logE(ERROR, "Empty or invalid frequency list; disabling transmission.");
         config.transmit = false;
         return false;

--- a/src/arg_parser.cpp
+++ b/src/arg_parser.cpp
@@ -885,7 +885,7 @@ bool parse_command_line(int argc, char *argv[])
     while (true)
     {
         int option_index = 0;
-        int c = getopt_long(argc, argv, "h?vnrDp:x:t:a:l:s:d:w:k:", long_options, &option_index);
+        int c = getopt_long(argc, argv, "h?vnroDp:x:t:a:l:s:d:w:k:", long_options, &option_index);
 
         if (c == -1)
             break;

--- a/src/scheduling.cpp
+++ b/src/scheduling.cpp
@@ -454,12 +454,25 @@ bool wspr_loop()
         callback_transmission_complete);
 
     // Wait for something to happen
+
     llog.logS(INFO, "WSPR loop running.");
 
     // Set pending config flags and do initial config
     ini_reload_pending.store(true, std::memory_order_relaxed);
     ppm_reload_pending.store(true, std::memory_order_relaxed);
-    set_config(true); // Handles get next (or only) frequency, PPM, and setup
+    if (config.mode == ModeType::WSPR)
+    {
+        // Set up WSPR transmissions
+        set_config(true); // Handles get next (or only) frequency, PPM, and setup
+    }
+    else
+    {
+        // Setup test tone
+        validate_config_data();
+        wsprTransmitter.setupTransmission(config.test_tone, config.power_level, config.ppm);
+        wsprTransmitter.enableTransmission();
+        llog.logS(INFO, "Transitting tone, hit Ctrl-C to terminate tone.");
+    }
 
     // -------------------------------------------------------------------------
     // Loop (block wspr_loop only) until shutdown is triggered

